File: TumeKyouen/Dao/BaseDao.m
//
//  BaseDao.m
//  TumeKyouen
//
//  Created by 石倉 昇 on 2013/03/03.
//  Copyright (c) 2013年 noboru. All rights reserved.
//

#import "BaseDao.h"
#import "TKAppDelegate.h"

@implementation BaseDao

@synthesize managedObjectContext;

- (id)init {
    if (self = [super init]) {
        TKAppDelegate *appDelegate =
            (TKAppDelegate *)[[UIApplication sharedApplication] delegate];
        managedObjectContext = [appDelegate managedObjectContext];
    }
    return self;
}

@end

File: TumeKyouen/Dao/BaseDao.swift
//
//  BaseDao.swift
//  TumeKyouen
//
//  Created by 石倉 昇 on 2016/02/03.
//  Copyright © 2016年 noboru. All rights reserved.
//

import UIKit
import CoreData

class BaseDao {
    lazy var managedObjectContext: NSManagedObjectContext = {
        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else {
            abort()
        }
        return appDelegate.managedObjectContext
    }()
}
